pkg/cosmos/client.go

```diff
@@ -0,0 +1,95 @@
+package cosmos
+
+import (
+	"bytes"
+	"encoding/json"
+	"net/url"
+
+	"github.com/dcos/dcos-cli/pkg/httpclient"
+)
+
+// Client is a client for Cosmos.
+type Client struct {
+	http *httpclient.Client
+}
+
+// NewClient creates a new Cosmos client.
+func NewClient(baseClient *httpclient.Client) *Client {
+	return &Client{
+		http: baseClient,
+	}
+}
+
+// PackageInfo contains information about a DC/OS package.
+// It is mirroring the JSON response of the `/package/describe`
+// endpoint for unmarshalling convenience.
+type PackageInfo struct {
+	Package struct {
+		Name        string `json:"name"`
+		Description string `json:"description"`
+		Version     string `json:"version"`
+		Resource    struct {
+			CLI struct {
+				Plugins map[string]map[string]*Plugin `json:"binaries"`
+			} `json:"cli"`
+		} `json:"resource"`
+	} `json:"package"`
+}
+
+// Plugin represents a CLI plugin resource.
+type Plugin struct {
+	Kind string `json:"kind"`
+	URL  string `json:"url"`
+}
+
+// DescribePackage returns information about the named package.
+func (c *Client) DescribePackage(name string) (*PackageInfo, error) {
+	reqBodyPayload := map[string]string{"packageName": name}
+
+	var reqBody bytes.Buffer
+	if err := json.NewEncoder(&reqBody).Encode(reqBodyPayload); err != nil {
+		return nil, err
+	}
+
+	req, err := c.http.NewRequest("POST", "/package/describe", &reqBody, httpclient.FailOnErrStatus(true))
+	if err != nil {
+		return nil, err
+	}
+	req.Header.Set(
+		"Content-Type",
+		"application/vnd.dcos.package.describe-request+json;charset=utf-8;version=v1",
+	)
+	req.Header.Set(
+		"Accept",
+		"application/vnd.dcos.package.describe-response+json;charset=utf-8;version=v3",
+	)
+	resp, err := c.http.Do(req)
+	if err != nil {
+		return nil, err
+	}
+
+	defer resp.Body.Close()
+
+	var pkg PackageInfo
+	err = json.NewDecoder(resp.Body).Decode(&pkg)
+	if err != nil {
+		return nil, err
+	}
+
+	// Workaround for a Cosmos bug leading to wrong schemes in plugin resource URLs.
+	// See https://jira.mesosphere.com/browse/COPS-3052 for more context.
```

Comment:
The COPS ticket is closed and cannot be opened by devs outside of Mesosphere, a description of the problem handling this test case check would be more useful.

Reply:
I've added more comments.